ptcs/ptcs_control/control.py

```diff
@@ -79,6 +79,69 @@ def move_train(self, train_id: "Train", delta: float) -> None:
             train_state.current_section = next_section
             train_state.target_junction = next_target_junction
 
+    def calc_direction(self) -> None:
+        """
+        ポイントをどちら向きにするかを計算する。
+        """
+
+        # junction定義
+        j0a = Junction("j0a")
+        j0b = Junction("j0b")
+        j1a = Junction("j1a")
+        j1b = Junction("j1b")
+        # sectionの定義
+        s1 = Section("s1")
+        s2 = Section("s2")
+        s3 = Section("s3")
+        s5 = Section("s5")
+        # 「とりうるルート」の列挙
+        possible_junction_direction: dict[str, list(tuple[Junction, Direction])] = {
```

Comment:
`list(...)` → `list[...]` かも

```suggestion
        possible_junction_direction: dict[str, list[tuple[Junction, Direction]]] = {
```

Reply:
型ヒントに疎過ぎた、反映しましたありがとう！